code/obj/mining_cloud_storage.dm

```diff
@@ -0,0 +1,264 @@
+			var/amtload = 0
+			for (var/obj/item/M in O.contents)
+				if (!istype(M,src.base_material_class))
+					continue
+				src.load_item(M)
+				amtload++
+			if (amtload)
+				boutput(user, "<span class='notice'>[amtload] materials loaded from [O]!</span>")
+				src.update_ores()
+			else boutput(user, "<span class='alert'>No material loaded!</span>")
+
+		else if (isitem(O))
+			quickload(user,O)
+			src.update_ores()
+
+		else
+			boutput(world,"[O.type] is O type")
+			if(isitem(O))
+				boutput(world,"[O] is item")
+			..()
+
+		src.updateUsrDialog()
+
+	proc/quickload(var/mob/living/user,var/obj/item/O)
+		if (!user || !O)
+			return
+		user.visible_message("<span class='notice'>[user] begins quickly stuffing [O] into [src]!</span>")
+		var/staystill = user.loc
+		for(var/obj/item/M in view(1,user))
+			if (!M)
+				continue
+			if (M.name != O.name)
+				continue
+			if (!istype(M,src.base_material_class))
+				continue
+			if (O.loc == user)
+				continue
+			if (O in user.contents)
+				continue
+				src.load_item(M)
+			M.set_loc(src)
+			playsound(get_turf(src), sound_load, 40, 1)
+			sleep(0.5)
+			if (user.loc != staystill) break
+		boutput(user, "<span class='notice'>You finish stuffing [O] into [src]!</span>")
+		return
+
+	attackby(obj/item/W as obj, mob/user as mob)
+		if (istype(W, src.base_material_class) && src.accept_loading(user))
+			user.visible_message("<span class='notice'>[user] loads [W] into the [src].</span>", "<span class='notice'>You load [W] into the [src].</span>")
+			src.load_item(W,user)
+			src.update_ores()
+
+		else
+			src.health = max(src.health-W.force,0)
+			src.check_health()
+			..()
+
+	proc/check_health()
+		if(!src.health)
+			src.broken = 1
+			src.icon_state = "ore_storage_unit-broken"
+			src.update_sellable()
+
+	proc/load_item(var/obj/item/O,var/mob/living/user)
+		if (!O)
+			return
+		if(O.amount == 1)
+			O.set_loc(src)
+			if (user && O)
+				user.u_equip(O)
+				O.dropped()
+		else if(O.amount>1)
+			O.set_loc(src)
+			for(O.amount,O.amount > 0, O.amount--)
+				new O.type(src)
+			if (user && O)
+				user.u_equip(O)
+				O.dropped()
+			qdel(O)
+		else
+			return // uhhhhhh
+
+
+	proc/accept_loading(var/mob/user,var/allow_silicon = 0)
+		if (!user)
+			return 0
+		if (src.status & BROKEN || src.status & NOPOWER)
+			return 0
+		if (!istype(user, /mob/living/))
+			return 0
+		if (istype(user, /mob/living/silicon) && !allow_silicon)
+			return 0
+		var/mob/living/L = user
+		if (!isalive(L) || L.transforming)
+			return 0
+		return 1
+
+	proc/update_ores()
+		src.ores.len = 0
+		for(var/obj/item/raw_material/R in src.contents)
+			if(!(R.material_name in ores)) //Using material_name to group materials is dangerous in general. However, since we are only accepting raw_materials, this should be fine.
+				ores += R.material_name
```

Comment:
```suggestion
			if(!ores[R.material_name]) //Using material_name to group materials is dangerous in general. However, since we are only accepting raw_materials, this should be fine.
```

Reply:
Added in pending commit